safekeeper/src/send_interpreted_wal.rs

```diff
@@ -287,7 +287,13 @@ impl InterpretedWalReader {
                 reader
                     .run_impl(start_pos)
                     .await
-                    .inspect_err(|err| critical!("failed to read WAL record: {err:?}"))
+                    .inspect_err(|err| match err {
+                        // TODO: we may want to differentiate these errors further.
+                        InterpretedWalReaderError::Decode(_) => {
+                            critical!("failed to decode WAL record: {err:?}");
+                        }
```

Comment:
We should split up `InterpretedWalReaderError::ReadOrInterpret` into `Read` and `Interpret`.
Interpretation errors should be critical.

Reply:
Yeah, but that's going to require a bunch of plumbing since `StreamingWalReader` and all the inner infra currently uses `anyhow::Error`. I'll merge this as-is to get the downgrade in. Are you up for some plumbing? :)